src/Cryptol/Parser/ParserUtils.hs

```diff
@@ -1010,39 +1008,73 @@ mkDoc ltxt = ltxt { thing = docStr }
   where
 
   docStr = T.unlines
-         $ dropPrefix
-         $ trimFront
+         $ handlePrefixes
          $ T.lines
          $ T.dropWhileEnd commentChar
          $ thing ltxt
 
   commentChar :: Char -> Bool
-  commentChar x = x `elem` ("/* \r\n\t" :: String)
+  commentChar x = x `elem` ("/* \t" :: String)
 
-  prefixDroppable x = x `elem` ("* \r\n\t" :: String)
+  prefixDroppable :: Char -> Bool
+  prefixDroppable x = x `elem` ("* \t" :: String)
 
   whitespaceChar :: Char -> Bool
-  whitespaceChar x = x `elem` (" \r\n\t" :: String)
-
-  trimFront []                     = []
-  trimFront (l:ls)
-    | T.all commentChar l = ls
-    | otherwise           = T.dropWhile commentChar l : ls
-
-  dropPrefix []        = []
-  dropPrefix [t]       = [T.dropWhile commentChar t]
-  dropPrefix ts@(l:ls) =
+  whitespaceChar x = x `elem` (" \t" :: String)
```

Comment:
Why not use something like [`Data.Char.isSpace`](https://hackage.haskell.org/package/base-4.20.0.1/docs/Data-Char.html#v:isSpace) to implement this?

Reply:
I think even letting tab get mixed in will create some really confusing situations but I'm happy to allow is space to really let people go wild